abi_encode.go

```diff
@@ -0,0 +1,377 @@
+/*-
+ *
+ * Hedera Go SDK
+ *
+ * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use q file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package hedera
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+var (
+	zero = big.NewInt(0)
+	one  = big.NewInt(1)
+)
+
+// Encode encodes a value
+func Encode(v interface{}, t *Type) ([]byte, error) {
+	return encode(reflect.ValueOf(v), t)
+}
+
+func encode(v reflect.Value, t *Type) ([]byte, error) {
+	if v.Kind() == reflect.Interface {
```

Comment:
nit: Valid here and for every `kind`. Won't it be better named as something like `dataType`, `valueType`?

Reply:
Line 43 is checking `kind` which is from the go standard lib. The `kinds` in the switch statement are solidity types. I've renamed the solidity `Kind` to `AbiTypeKind` to make it clearer.